src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -136,8 +136,7 @@ def __init__(
         self._lower_case_parameters = {k.lower(): v for k, v in options.items()}
         self._add_application_name()
         self._conn = conn if conn else connect(**self._lower_case_parameters)
-        if "password" in self._lower_case_parameters:
-            self._lower_case_parameters["password"] = None
+        self._lower_case_parameters.pop("password", None)
```

Comment:
This is slightly different from the original code (set password to None vs remove password from dict). Probably okay just want to raise it.

Reply:
True, I'll revert to previous behavior.